README.md

```diff
@@ -317,81 +317,52 @@ the content as below:
 
 #### RichDateInput
 
-The implicit class `RichDateInput` provides an extension method `withFormField(field: play.api.data.Field)`
-for the `DateInput` view model.
-
-This method takes a Play `Field` and enriches the `DateInput` with:
-* three InputItems corresponding to the day, month and year fields with
-    * ids and names of the form `<name>.day`, `<name>.month` and `<name>.year`
-    * standard English and Welsh labels
+The implicit class `RichDateInput` provides three extension methods to populate the `DateInput` view model:
+* `withDayMonthYearFormField(field: Field)`
+* `withDayMonthFormField(field: Field)`
+* `withMonthYearFormField(field: Field)`
+
+These methods takes a Play `play.api.data.Field` and enrich the `DateInput` with:
+* InputItems corresponding to a combination of the day, month and year fields with
+    * ids and names of the form `<name>.day`, `<name>.month` and `<name>.year`  (`<name>` being populated from the `Field` name if present, else `date`).
+* standard English and Welsh labels
 * an id, set to the `Field` name
 * values for the nested day, month and year fields
 * a Text error message from an implicit `Messages` indexed by the first non-empty:
-    * `field("day").error`
-    * `field("month").error`
-    * `field("year").error`
-    * `field.error`
-* the correct CSS error classes applied to any invalid nested day, month or year field or
-  to all fields in the case of a global date error.
-
-This helper assumes the Field passed to it consists of three [nested fields](https://www.playframework.com/documentation/2.8.x/ScalaForms#Nested-values)
-indexed by the keys `day`, `month` and `year`.
+* `field("day").error`
+* `field("month").error`
+* `field("year").error`
+* `field.error`
+* the correct CSS error classes applied to any invalid nested day, month or year field or  to all fields in the case of a global date error.
 
 For example, if using the one question per page pattern, the method could be used as follows:
 
-```scala
-@govukDateInput(DateInput(
-  hint = Some(Hint(content = Text("date.hint"))),
-  fieldset = Some(Fieldset(
-    legend = Some(Legend(
-      content = Text(messages("date.heading")),
-      classes = "govuk-fieldset__legend--l",
-      isPageHeading = true)))
-  )
-).withFormField(dateInputForm("date")))
-```
+```scala  
+@govukDateInput(DateInput(  
+ hint = Some(Hint(content = Text("date.hint"))), fieldset = Some(Fieldset( legend = Some(Legend( content = Text(messages("date.heading")), classes = "govuk-fieldset__legend--l", isPageHeading = true))) )).withDayMonthYearFormField(dateInputForm("date")))  
+```  
```

Comment:
I think it might be worth formatting these two code samples with line breaks again

Reply:
@oscarduignan Sure thing 🙌

Reply:
@oscarduignan Done